firmware/src/hal/sgx/src/trusted/secret_store.c

```diff
@@ -48,18 +49,64 @@
 
 #define MAX_BLOB_SIZE (1024 * 1024)
 
+#define SECRET_PAYLOAD_SIZE(secret_size) (secret_size - sizeof(secret_header_t))
+#define SEST_HASH_SIZE (32)
+
 // To avoid performing dynamic memory allocation whenever we need to read a
 // secret, we use this static buffer to temporarily store the sealed blobs we
 // read from the untrusted modules.
 static uint8_t G_secrets_buffer[MAX_BLOB_SIZE];
 
+// The sha256 context used for hash operations.
+static SHA256_CTX G_sha256_ctx;
+
+// The blob format of the sealed secrets we send to the seal API.
+// This includes the payload and a sha256 hash of the key for validation
+// purposes.
+typedef struct {
+    uint8_t key_hash[SEST_HASH_SIZE];
+} secret_header_t;
+
+typedef struct {
+    secret_header_t header;
+    uint8_t payload[0];
+} secret_t;
+
```

Comment:
I'm a bit concerned about alignment here. Can we guarantee this is held in memory as contiguous bytes?